File: lib/features/tasks/data/models/project.dart
import 'package:isar/isar.dart';

part 'project.g.dart';

@collection
class Project {
  Id id = Isar.autoIncrement;

  late String title;
  
  String? description;
  
  String? notes;
  
  double progress = 0.0;
  
  DateTime createdAt = DateTime.now();
}

File: backend/models/Project.js
const mongoose = require('mongoose');

const projectSchema = new mongoose.Schema({
  id: { type: Number, required: true },
  userId: { type: String, ref: 'User', required: true },
  title: { type: String, required: true },
  description: { type: String },
  notes: { type: String },
  progress: { type: Number, default: 0.0 },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Project', projectSchema);
